rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/db/sql/DBSQLFile.kt

```diff
@@ -31,29 +32,45 @@ class DBSQLFile(private val name: String, private val connection: Connection) :
     }
 
     override fun readEqual(key: Value): Record {
-        if (resultSet == null) {
-            chain(emptyList())
+        return readEqual(toFields(key))
+    }
+
+    private fun filterRecord(result: Record): Record {
+        return if (result.matches(lastKey)) {
+            result
+        } else {
+            resultSet?.last()
+            Record()
         }
-        return readFromPositionedResultSet()
     }
 
     override fun readEqual(keys: List<Field>): Record {
-        TODO("not implemented")
+        val result = if (resultSet == null) {
+            chain(emptyList())
+        } else {
+            readFromPositionedResultSet()
+        }
+        lastKey = keys
+        return filterRecord(result)
     }
 
     override fun eof(): Boolean = resultSet?.isLast ?: false
 
     override fun chain(key: Value): Record {
+        return chain(toFields(key))
+    }
+
+    private fun toFields(keyValue: Value): List<Field> {
```

Comment:
It is not clear to me what this function does. Probably because I am not familiar with this portion of the codebase

Reply:
A chain operation (lookup a record in a table) can be perfomed using one or more keys. So we have to transform a value (e.g. "ABC") in a Field (i.e. "PART_NUMBER"->"ABC") and then encapsulate  it in a list. Do you have any suggestion in order to make the code more clear?

Reply:
To me is not clear what a Field is in this context. Is it a data structure field? In your example is "PART_NUMBER" the name of the field used as a key and is "ABC" the actual value of the field?

Reply:
Also, the term Field makes me think to the data structure fields, while these are different fields, right? Perhaps we could have a more specific name to make this more obvious